File: RQ3/C-REval-Desc/task00_desc.c
#include <stdbool.h>
#include <math.h>

bool has_close_elements(double numbers[], int length, double threshold) {
    bool _result = false;
    for (int idx = 0; idx < length; idx++) {
        for (int idx2 = 0; idx2 < length; idx2++) {
            if (idx != idx2) {
                double distance = fabs(numbers[idx] - numbers[idx2]);
                if (distance < threshold) {
                    _result = true;
                }
            }
        }
    }
    return _result;
}

File: RQ3/Java-REval-Desc/task00_desc.java
public class Translator {
    public static boolean hasCloseElements(double[] numbers, double threshold) {
        boolean result = false;
        for (int idx = 0; idx < numbers.length; idx++) {
            for (int idx2 = 0; idx2 < numbers.length; idx2++) {
                if (idx != idx2) {
                    double distance = Math.abs(numbers[idx] - numbers[idx2]);
                    if (distance < threshold) {
                        result = true;
                    }
                }
            }
        }

        assert func(result); // POST: func is equivalent to: Check if in given list of numbers, are any two numbers closer to each other than given threshold. >>> has_close_elements([1.0, 2.0, 3.0], 0.5) False >>> has_close_elements([1.0, 2.8, 3.0, 4.0, 5.0, 2.0], 0.3) True
    }
}